src/flake8_tidy_imports/__init__.py

```diff
@@ -114,6 +139,36 @@ def rule_I250(
                         type(self),
                     )
 
+    @staticmethod
+    def compile_unstructured_glob(s: str) -> Pattern[str]:
+        # Convert the patter to a regex such that ".*"
+        # matches zero or more moduels.
+        parts = s.split(".")
+        transformed_parts = [
+            "(\\..*)?" if p == "*" else "\\." + re.escape(p) for p in parts
+        ]
+        if parts[0] == "*":
+            transformed_parts[0] = ".*"
+        else:
+            transformed_parts[0] = re.escape(parts[0])
+        return re.compile("".join(transformed_parts) + "\\Z")
+
+    def _is_module_banned(self, module_name: str) -> Tuple[bool, str]:
+        if module_name in self.banned_modules:
+            return True, self.banned_modules[module_name]
+
+        # Check unustructed wildcards
+        for banned_pattern, msg in self.banned_unstructured_patterns:
+            if banned_pattern.match(module_name):
+                return True, msg
+
+        # Check structured wildcards
+        for banned_prefix, msg in self.banned_structured_patterns:
```

Comment:
This is different than the approach in mypy:
https://github.com/python/mypy/blob/b3ff2a683bd41a176c597d3c3ba65f6a31f805c5/mypy/options.py#L396-L408
Their use case is slightly different since they want to cache the result.

Another approach would be to create a regex when parsing the options, so that we'd have one (bigger) list of regex's to deal with, but I prefer keeping them separate.